src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -1007,5 +1006,74 @@ void M()
                     //         Func<string, string, string> func = (x!!, y) => x;
                     Diagnostic(ErrorCode.ERR_FeatureInPreview, "!!").WithArguments("parameter null-checking").WithLocation(7, 47));
         }
+
+        [Fact]
+        public void TestNullCheckedParameterUpdatesFlowState1()
+        {
+            var source =
+@"
+#nullable enable
+
+class Program
+{
+    string M(string? s!!) // 1
+    {
+        return s;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (6,22): warning CS8995: Nullable type 'string?' is null-checked and will throw if null.
+                //     string M(string? s!!) // 1
+                Diagnostic(ErrorCode.WRN_NullCheckingOnNullableType, "s").WithArguments("string?").WithLocation(6, 22));
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing generic types and attributes:
```csharp
static void F1<T>(T x1, T y1!!)
{
    x1.ToString();
    y1.ToString();
}
static void F2<T>([AllowNull] T x2, [AllowNull] T y2!!)
{
    x2.ToString();
    y2.ToString();
}
static void F3<T>([DisallowNull] T x3, [DisallowNull] T y3!!)
{
    x3.ToString();
    y3.ToString();
}
```

Reply:
Done.